contracts/strategies/lift/LiftStrategy.sol

```diff
@@ -0,0 +1,271 @@
+  *   Those are protected by the "unsalvagableTokens". To check, see where those are being flagged.
+  */
+  function salvage(address recipient, address token, uint256 amount) external onlyControllerOrGovernance {
+     // To make sure that governance cannot come in and take away the coins
+    require(!unsalvagableTokens[token], "token is defined as not salvagable");
+    IERC20(token).safeTransfer(recipient, amount);
+  }
+
+  /*
+  *   Get the reward, sell it in exchange for underlying, invest what you got.
+  *   It's not much, but it's honest work.
+  *
+  *   Note that although `onlyNotPausedInvesting` is not added here,
+  *   calling `investAllUnderlying()` affectively blocks the usage of `doHardWork`
+  *   when the investing is being paused by governance.
+  */
+  function doHardWork() external onlyNotPausedInvesting restricted {
+    uint256 len = stakes.length;
+    require(len<=200, "Too many stakes, withdraw first");
+    uint256 rewardBalanceBefore = boardRoom.getbalanceOfShare(address(this));
+    rewardPool.stakeInBoardroom();
+    uint256 rewardBalanceAfter = boardRoom.getbalanceOfShare(address(this));
+    uint256 rewardStaked = rewardBalanceAfter.sub(rewardBalanceBefore);
+    if (rewardStaked > 0) {
+      stakes.push([block.timestamp, rewardStaked]);
+    }
+    investAllUnderlying();
+  }
+
+  function _getReward() internal {
+    uint256 len = stakes.length;
+    require(len<=200, "Too many stakes, withdraw first");
+    uint256 rewardBalanceBefore = boardRoom.getbalanceOfShare(address(this));
+    rewardPool.stakeInBoardroom();
+    uint256 rewardBalanceAfter = boardRoom.getbalanceOfShare(address(this));
+    uint256 rewardStaked = rewardBalanceAfter.sub(rewardBalanceBefore);
+    if (rewardStaked > 0) {
+      stakes.push([block.timestamp, rewardStaked]);
+    }
+  }
+
+  function withdrawRewardShareOldest() external onlyMultiSigOrGovernance {
+    require(allowedRewardClaimable, "reward claimable is not allowed");
+    uint256 len = stakes.length;
+    require(len>0, "no stakes");
+
+    uint256 time = stakes[0][0];
+    boardRoom.withdrawShare(time);
+    uint256 shareBalance = IERC20(rewardToken).balanceOf(address(this));
+    IERC20(rewardToken).safeTransfer(msg.sender, shareBalance);
+
+    //clean up the list of stakes
+    uint256 i = 0;
+    for (i;i<(len.sub(1));i++) {
+      stakes[i] = stakes[i+1];
+    }
+    stakes.pop();
+  }
+
+  function withdrawRewardShareNewest() external onlyMultiSigOrGovernance {
+    require(allowedRewardClaimable, "reward claimable is not allowed");
+    uint256 len = stakes.length;
+    require(len>0, "no stakes");
+
+    uint256 time = stakes[len.sub(1)][0];
+    boardRoom.withdrawShare(time);
+    uint256 shareBalance = IERC20(rewardToken).balanceOf(address(this));
+    IERC20(rewardToken).safeTransfer(msg.sender, shareBalance);
+    stakes.pop();
+  }
+
+  function withdrawRewardShareAll() external onlyMultiSigOrGovernance {
```

Comment:
This method needs to have a way of providing the maximum length, otherwise at some point, this method may no longer work due to too many stakes in the stakes array.

Reply:
The stakes array size is limited to 200, I will make a variable with a change function for this instead, that makes more sense than hardcoding 200, but there is a size limit.